collectors/o365/o365_collector.js

```diff
@@ -73,29 +77,63 @@ class O365Collector extends PawsCollector {
     
     pawsGetLogs(state, callback) {
         let collector = this;
-        console.info(`O365000001 Collecting data from ${state.since} till ${state.until}`);
-        const newState = collector._getNextCollectionState(state);
+        console.info(`O365000001 Collecting data from ${state.since} till ${state.until} for stream ${state.stream}`);
+        let pageCount = 0;
 
-        // get the content for the timestamp
-        const streams = JSON.parse(process.env.O365_CONTENT_STREAMS)
-        const contentPromises = streams.map((stream) => {
-            return m_o365mgmnt.subscriptionsContent(stream, state.since, state.until)
-        });
-        // Get the streams from env vars and collect the content
-        Promise.all(contentPromises).then(results => {
-            const aggregateResult = results.reduce((agg, e) => [...e, ...agg], []);
-
-            const contentPromises = aggregateResult.map(({contentUri}) => m_o365mgmnt.getContent(contentUri));
-            return Promise.all(contentPromises);
-        // Call all of the content urls and agregate the resutls
-        }).then(content => {
-            const flattenedResult = content.reduce((agg, e) => [...e, ...agg], []);
-
-            console.info(`O365000002 Next collection in ${newState.poll_interval_sec} seconds`);
-            callback(null, flattenedResult, newState, newState.poll_interval_sec);
+        // Page aggregation handler
+        const listContentCallback = ({parsedBody, nextPageUri}) => {
+
+            if(nextPageUri && pageCount < process.env.maxPages){
+                pageCount++;
+                return m_o365mgmnt.getPreFormedUrl(nextPageUri)
+                    .then((nextPageRes) => {
+                        return {
+                            parsedBody: [...parsedBody, ...nextPageRes.parsedBody],
+                            nextPageUri: nextPageRes.nextPageUri
+                        }
+                    })
+                    .then(listContentCallback);
+            }
+            else{
+                return {
+                    parsedBody,
+                    nextPageUri
+                }
+            }
+        };
+
+        // If the state has a next page value, then just start with that.
+        const initialListContent = state.nextPage ?
+            contentPromise =  m_o365mgmnt.getPreFormedUrl(state.nextPage):
```

Comment:
do we actually need this assignment?
I see `contentPromise` var is defined below.